feat: use Nest.js SDK in backend

lukas-reining opened the pull request:
<!-- Please use this template for your pull request. -->
<!-- Please use the sections that you need and delete other sections -->

## This PR
<!-- add the description of the PR here -->

Adds the Nest.js SDK to the backend services.

beeme1mr commented on packages/app/src/app/app.module.ts:
```diff
@@ -1,50 +1,26 @@
-import { FlagMetadata, OpenFeature } from '@openfeature/server-sdk';
-import {MetricsHook, TracingHook as SpanEventBasedTracingHook} from '@openfeature/open-telemetry-hooks';
+import {LoggingHook, OpenFeatureLogger} from '@openfeature/extra';
+import {FlagMetadata} from '@openfeature/js-sdk';
```
I'm surprised this worked unless we didn't remove the `js-sdk` from thepackage.json. This should be exported from the nest sdk, please use that one.

beeme1mr commented on packages/app/src/app/fibonacci/fibonacci.service.ts:
```diff
@@ -1,15 +1,16 @@
+import {HttpService} from '@nestjs/axios';
+import {Injectable} from '@nestjs/common';
+import {fibonacci} from '@openfeature/fibonacci';
+import {Client} from '@openfeature/js-sdk';
```
Please use the nest sdk

beeme1mr commented on packages/fibonacci-service/src/app/app.module.ts:
```diff
@@ -1,40 +1,19 @@
 import {LoggerModule} from 'nestjs-pino';
-import { FlagMetadata, OpenFeature } from '@openfeature/server-sdk';
-import {AsyncLocalStorageTransactionContext, LoggingHook, OpenFeatureLogger} from '@openfeature/extra';
+import {FlagMetadata} from '@openfeature/server-sdk';
```
Please use `@openfeature/nestjs-sdk`.

beeme1mr reviewed (commented):
Nice, looks good. Were you able to run this locally? I'll try and test it out this afternoon.

lukas-reining commented:
Yes @beeme1mr, building it locally and running worked!
I will change the imports tomorrow.

lukas-reining commented:
I updated the imports @toddbaert @beeme1mr :)

beeme1mr reviewed (commented):
I'm running into an issue building the UI. It seems to have something to do with missing webpack polyfills. I'm looking into it.

beeme1mr commented:
> I'm running into an issue building the UI. It seems to have something to do with missing webpack polyfills. I'm looking into it.

I found the issue. There was a shared library that was used both on the client and server but was importing from the Nest SDK. I've updated the imports and added a linting rule to prevent it from happening in the future.

lukas-reining commented:
> > I'm running into an issue building the UI. It seems to have something to do with missing webpack polyfills. I'm looking into it.
> 
> 
> 
> I found the issue. There was a shared library that was used both on the client and server but was importing from the Nest SDK. I've updated the imports and added a linting rule to prevent it from happening in the future.

Mh interesting, I could build it with the former `npm run temp` and thought I cleaned everything before 🤔
Thank you for fixing it!
